### DISCUSSION TITLE
Change default babystep values

### DISCUSSION THREAD

**Discussion by beffjessen** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2700] [Q&A] [answered]
Is it possible to change the default toggle values for babystepping in touchscreen mode? I recently added a TFT35 E3 V3.0.1 to my Ender 3 Pro, and while it's a huge improvement over the stock screen, the options for babystepping while printing don't seem ideal. With the stock screen, I could babystep in increments of 0.025mm, while the TFT35 can only be set at 0.01, 0.1, or 1mm. 0.01mm feels way too granular, and a 1mm adjustment feels like it's pushing the definition of "baby" :) - which leaves 0.1mm as the only somewhat usable step size, although that's often too blunt. I tried skimming through the firmware source files to see where those default babystep values are defined, and came up empty (that said, I'm very much a MARLIN/C newbie). I appreciate any feedback on whether this is possible - thanks!

**Comment by kisslorand** [comment, marked as answer]
You can adjust your own values in "Common.c" at line 75.
`const float moveLenSteps[ITEM_MOVE_LEN_NUM] = {0.01f, 0.1f, 1, 10, 100};`

**Reply by beffjessen** [reply]
Amazing, thank you! Two follow-up questions:
- Is it possible to show the babystep / Z offset to 3 decimal places?
- How do I change the labels for the new toggle steps that I set? For example, the 3 values I set are 0.01, 0.025, and 0.1, and the machine moves those distances when I send the command from the touchscreen, but the touchscreen shows 0.01, 0.1, and 1mm still (both the label and the icon). I know I would need to create new icons (if I care enough) but is there a simple way to change the labels at least?

**Comment by kisslorand** [comment]
Check in the "language_xx.h" file of your choice the following:
```
    #define STRING_001_MM                 "0.01mm"
    #define STRING_01_MM                  "0.1mm"
    #define STRING_1_MM                   "1mm"
    #define STRING_5_MM                   "5mm"
    #define STRING_10_MM                  "10mm"
    #define STRING_100_MM                 "100mm"
    #define STRING_200_MM                 "200mm"
```

**Reply by beffjessen** [reply]
Thanks for pointing me in the right direction. This ended up being somewhat involved to do - listing my steps below, in case it helps anyone in the future:
- Added new desired value to line 75 in "Common.c":
```
const float moveLenSteps[ITEM_MOVE_LEN_NUM] = {0.01f, 0.025f, 0.1f, 1, 10}
```
- Immediately above in "Common.c", changed the item list for distance change steps:
```
// Icons list for axis length/distance change steps
const ITEM itemMoveLen[ITEM_MOVE_LEN_NUM] =
{
// icon                          label
  {ICON_001_MM,                  LABEL_001_MM},
  {ICON_RESET_VALUE,             LABEL_025_MM}, // Note I borrowed a generic icon for the new value
  {ICON_01_MM,                   LABEL_01_MM},
  {ICON_1_MM,                    LABEL_1_MM},
  {ICON_10_MM,                   LABEL_10_MM},
};
```
- Created corresponding keywords in both "Language.inc" and "Language_keywords.h" under line 215, as well as in "language_xx.h" under line 208

I wasn't able to find a way to show a third decimal for the Z offset.
